test/functional/api/v1/reports_controller_test.rb

```diff
@@ -69,4 +71,25 @@ class Api::V1::ReportsControllerTest < ActionController::TestCase
     get :last, {:host_id => host.to_param }
     assert_response :not_found
   end
+
+  context 'when user does not have permission to view hosts' do
+    setup    { setup_user 'view', 'reports' }
+    teardown { assert_response :not_found }
```

Comment:
I don't think teardown is the right place for a behaviour assertion, better to just copy it into each test block.